tests/test_external_strategies.py

```diff
@@ -0,0 +1,119 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+from unittest.mock import MagicMock, Mock
+
+from d3a.models.area import Area
+from d3a.models.strategy.external_strategies.load import LoadForecastExternalStrategy
+from d3a.models.strategy.external_strategies.pv import PVForecastExternalStrategy
+from d3a_interface.constants_limits import GlobalConfig
+from pendulum import now, Duration
+from pytest import mark
+
+
+def ext_strategy_fixture(strategy):
+    config = MagicMock()
+    config.ticks_per_slot = 90
+    GlobalConfig.end_date = GlobalConfig.start_date + Duration(days=1)
+    area = Area(name="forecast_pv", config=config, strategy=strategy,
+                external_connection_available=True)
+    parent = Area(name="parent_area", children=[area])
+    parent.activate()
+    area.activate()
+    strategy.connected = True
+    market = MagicMock()
+    market.time_slot = GlobalConfig.start_date
+    parent.get_future_market_from_id = lambda _: market
+    area.get_future_market_from_id = lambda _: market
+    return strategy
+
+
+class TestPVForecastExternalStrategy:
+
+    @mark.parametrize("strategy", [ext_strategy_fixture(LoadForecastExternalStrategy()),
+                                   ext_strategy_fixture(PVForecastExternalStrategy())])
+    def test_clear_energy_buffers(self, strategy):
+        strategy.energy_forecast_buffer = {now(): 1}
+        strategy.energy_measurement_buffer = {now(): 1}
+        strategy._clear_energy_buffers()
+        assert strategy.energy_forecast_buffer == {}
+        assert strategy.energy_measurement_buffer == {}
+
+    @mark.parametrize("strategy", [ext_strategy_fixture(LoadForecastExternalStrategy()),
+                                   ext_strategy_fixture(PVForecastExternalStrategy())])
+    def test_update_energy_forecast_measurement_is_called_on_market_cycle(self, strategy):
```

Comment:
Nit: this test works and it's good to have it. I was just a bit mislead by the test name, because here you're actually testing `event_market_cycle`, not `update_energy_forecast/measurement`. 
One popular way to name test methods is to follow a similar schema: 
> `[MethodUnderTest]_[Scenario]_[ExpectedResult]`

For example: `test_event_market_cycle_calls_energy_update_methods()`. This is good because it immediately tells you which method you are testing, and for what. It makes it much easier to find tests for one method by using `test_<method-name>` in your code.

[In reality, you might want to test the entire `event_market_cycle` method (not just these calls) and that's one of the reasons why it's recommended to keep methods very short.]

Reply:
Understood and agreed. Thanks! Will change it.